### PR TITLE
Add RHEL-9.5 repo definitions

### PR DESCRIPTION
**Pull request by thozza**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the READMEs [listed here](https://github.com/osbuild/osbuild.github.io/blob/main/readme-list)
  - [ ] submit a PR for the [osbuild.org website](https://github.com/osbuild/osbuild.github.io/) repository if this PR changed any behavior not covered by the automatically updated READMEs

### CODE REVIEW

**Review by achilleas-k — DISMISSED** [review]

**Comment by thozza** [comment]
Sorry, I forgot to also add the 9.5 snapshot repos used for testing... :innocent:

**Review by achilleas-k — APPROVED** [review]

File: test/data/repositories/rhel-9.5.json [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      "name": "google-cloud-sdk",
+      "baseurl": "https://rpmrepo.osbuild.org/v2/mirror/public/el9/el9-x86_64-google-cloud-sdk-20240515/",
+      "check_gpg": false,
+      "image_type_tags": [
+        "gce",
+        "gce-rhui"
+      ]
+    }
+  ]
+}
```
Comment (croissanne) [inline test/data/repositories/rhel-9.5.json:146]: Mini-nitpick: trailing newline.

**Review by croissanne — COMMENTED** [review]
Mini-nitpick + ig the schutzfile changes.

File: test/data/repositories/rhel-9.5.json (hunk 1 above)
Comment (thozza) [reply test/data/repositories/rhel-9.5.json:146]: I just copied the `rhel-9.4.json` and replaced `9.4` with `9.5` :innocent:

Comment (achilleas-k) [reply test/data/repositories/rhel-9.5.json:146]: I get this a lot when I edit the files manually too.  Schutzbot uses `json.dump()` which doesn't add a newline, but then most editors will add it when saved, so it's a constant battle.

**Comment by thozza** [comment]
> Mini-nitpick + ig the schutzfile changes.

@croissanne  Can you be more specific on the schutzfile changes? I looked at it and the only thing that could make sense is to pin the osbuild version for 9.5, but I don't think that's needed at this point. 9.5 is getting the latest osbuild releases and if some change requires the version to be pinned, it should be done as part of it.

**Pull request merged**